tests/filesystem.go

```diff
@@ -187,8 +186,11 @@ func setDevices(t *testing.T, ctx context.Context, imageFile string, partitions
 	}
 	loopDevice := strings.TrimSpace(string(out))
 
-	// To avoid races with the kernel with loop devices appearing, sleep a bit
-	time.Sleep(time.Millisecond * 100)
+	// Avoid race with kernel by waiting for loopDevice creation to complete
+	_, err = run(t, ctx, "udevadm", "settle")
+	if err != nil {
+		return "", err
```

Comment:
nit: let's update the err to something like `fmt.Errorf("settling devices: %v", err)`